cdap-docs/reference-manual/source/release-notes.rst

```diff
@@ -31,6 +31,68 @@ Cask Data Application Platform Release Notes
    :backlinks: none
    :depth: 2
 
+
+`Release 4.1.2 <http://docs.cask.co/cdap/4.1.2/index.html>`__
+=============================================================
+
+Improvements
+------------
+
+- :cask-issue:`CDAP-12020` - Reuse network connections for TMS client.
+
+- :cask-issue:`CDAP-11959` - Adds a way to limit the frequency of retrieving the MapReduce task report, which could cause network load for very large jobs.
+
+- :cask-issue:`CDAP-11949` - Adds the ability to configure the HBase client scanner cache for a dataset.
+
+- :cask-issue:`CDAP-11594` - Added startup check for CDAP master to error out if the configurations for HBaseDDLExecutor extensions are provided, however extension jar cannot be loaded.
+
+- :cask-issue:`CDAP-11444` - Upgraded IDEA IntelliJ IDE in CDAP SDK VM to 2017.1.3 release.
+
+- :cask-issue:`CDAP-11398` - Upgraded Eclipse IDE in CDAP SDK VM to Neon 3 release.
+
+- :cask-issue:`CDAP-9515` - Added the ability to denormalize data, by splitting based on de-limiter text or array flattening, to individual records in Dataprep UI as point and click directive.
+
+- :cask-issue:`CDAP-9514` - Added the ability to apply some DataPrep directives on multiple columns, starting with Join columns and Swap columns. Multiple columns can be selected by checking the checkbox next to each column's name, then selecting a directive in the directive dropdown.
+
+- :cask-issue:`CDAP-9507` - Added the ability to format data (date time, string formatting etc.,) in Dataprep UI as point and click directive.
+
+- :cask-issue:`CDAP-9523` - Added the ability to extract text using regex patterns in Dataprep UI as point and click directive.
+
+- :cask-issue:`CDAP-9096` - Added feature where macro arguments are also listed in the runtime arguments of preview mode, just like when running a new pipeline.
+
+- :cask-issue:`CDAP-9094` - Added feature where values of macro arguments are automatically populated and shown in the UI when running a pipeline, if those values exist as Preferences.
+
+- :cask-issue:`CDAP-6329` - Enable GC logging for cdap services.
+
+
+Bug Fixes
+---------
+
+- :cask-issue:`CDAP-11985` - Fixed a bug that ugi provider returns the old and incorrect ugi information.
+
+- :cask-issue:`CDAP-11955` - Fixed a bug that sometimes wrong user is used in explore, which results in the failure of deleting namespace.
+
+- :cask-issue:`CDAP-11948` - Fix a bug where committed data could be removed during HBase table flush or compaction.
```

Comment:
I think we should be consistent with the tense (different notes use different forms: Fix, Fixes, Fixed).